relay-system/src/controller.rs

```diff
@@ -120,26 +115,26 @@ impl Controller {
     /// returns an error, the actix system is not started and instead an error returned. Otherwise,
     /// the system blocks the current thread until a shutdown signal is sent to the server and all
     /// actors have completed a graceful shutdown.
-    pub fn run<F, R, E>(factory: F) -> Result<(), E>
+    pub fn run<F, R, E>(shutdown_timeout: Duration, factory: F) -> Result<(), E>
     where
-        F: FnOnce() -> Result<R, E> + 'static,
-        F: Sync + Send,
+        F: FnOnce() -> Result<R, E>,
     {
         // Spawn a legacy actix system for the controller's signals.
         let sys = actix::System::new("relay");
 
-        // Ensure that the controller starts if no service has started it yet. It will register with
+        // Ensure that the controller starts if no service has starts it. It will register with
```

Comment:
```suggestion
        // Ensure that the controller starts if no service has started it yet. It will register with
```

I think we should revert the comment to previous state.